crates/ruff_linter/src/rules/flake8_simplify/rules/yoda_conditions.rs

```diff
@@ -78,18 +80,51 @@ impl Violation for YodaConditions {
     }
 }
 
-/// Return `true` if an [`Expr`] is a constant or a constant-like name.
-fn is_constant_like(expr: &Expr) -> bool {
+/// Comparisons left hand side must not be more [`ConstantLikelihood`] than their right hand side
+#[derive(PartialEq, Eq, PartialOrd, Ord, Debug)]
+enum ConstantLikelihood {
+    Unlikely = 0,
+    Probably = 1,   // CAMEL_CASED vars
+    Definitely = 2, // literals, empty dicts and tuples...
```

Comment:
Nit: make these rustdocs? Like:

```rust
/// The expression is unlikely to be a constant (e.g., `foo` or `foo(bar)`).
Unlikely = 0
/// The expression is likely to be a constant (e.g., `FOO`).
Probably = 1,
...
```

Reply:
done. feel free to correct my English there...